File: rust/src/models/metric_dimension.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct MetricDimension {
    pub key: String,

    pub values: Vec<String>,
}

impl MetricDimension {
    pub fn new(key: String, values: Vec<String>) -> Self {
        Self { key, values }
    }
}

File: typescript/src/models/metricDimension.ts
// this file is @generated

export interface MetricDimension {
  key: string;

  values: string[];
}

export const MetricDimensionSerializer = {
  _fromJsonObject(object: any): MetricDimension {
    return {
      key: object["key"],
      values: object["values"],
    };
  },

  _toJsonObject(self: MetricDimension): any {
    return {
      key: self.key,
      values: self.values,
    };
  },
};
